### PR TITLE
Cse sum participation factors different to one

### PR DESCRIPTION
**Pull request by schwitzguebel**
**Please check if the PR fulfills these requirements**

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

**What kind of change does this PR introduce?**

**What is the current behavior?**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

### CODE REVIEW

File: glsk/glsk-document-io-api/src/main/java/com/powsybl/glsk/api/io/GlskDocumentImporter.java [hunk 1]
```diff
@@ -26,5 +26,9 @@ default GlskDocument importGlsk(InputStream inputStream, boolean useCalculationD
         throw new NotImplementedException("This importer does not handle CalculationDirections");
     }
 
+    default GlskDocument importAndValidateGlsk(InputStream inputStream, boolean useCalculationDirections) {
```
Comment (vbochetRTE) [inline glsk/glsk-document-io-api/src/main/java/com/powsybl/glsk/api/io/GlskDocumentImporter.java:29]: You probably should add a new method in the `GlskDocumentImporters` class to access this new `importAndValidateGlsk()` method, as I did for `importGlskWithCalculationDirections()` method.

File: glsk/glsk-document-cse/src/main/java/com/powsybl/glsk/cse/CseGlskPoint.java [hunk 2]
```diff
@@ -35,41 +36,53 @@ public CseGlskPoint(TimeSeriesType timeSeries) {
         BusinessTypeList businessType = timeSeries.getBusinessType().getV();
 
         try {
+            //pretreatment to set sum of participation factors of all blocks to 1 (100%)
+
+            //calculate sum of block factors
+            BigDecimal sumBlockFactors = Stream.concat(Stream.ofNullable(timeSeries.getManualLSKBlockOrPropLSKBlock()),
```
Comment (vbochetRTE) [inline glsk/glsk-document-cse/src/main/java/com/powsybl/glsk/cse/CseGlskPoint.java:42]: Maybe putting all this `Stream` operation in a well-named dedicated method would make the code easier to understand and make the comment above unnecessary.

File: glsk/glsk-document-cse/src/main/java/com/powsybl/glsk/cse/CseGlskShiftKey.java [hunk 3]
```diff
@@ -99,8 +99,15 @@ private void initCommonMemberVariables(BlockWrapper blockWrapper, BusinessTypeLi
             blockWrapper.getMaximumShift().map(BigDecimal::doubleValue)
                 .ifPresent(ms -> this.maximumShift = ms);
         } else {
-            blockWrapper.getFactor().map(BigDecimal::doubleValue)
-                .ifPresent(q -> this.quantity = q);
+            blockWrapper.getFactor()
```
Comment (vbochetRTE) [inline glsk/glsk-document-cse/src/main/java/com/powsybl/glsk/cse/CseGlskShiftKey.java:102]: From my point of view, putting the comparison of `sumBlockFactors` with 0 and 1 in a final variable allows us to write some clearer code. You may not share the same opinion :smiley:.

A proposal below:
```java
            final boolean sumIsZeroOrOne =
                BigDecimal.ZERO.compareTo(sumBlockFactors) == 0
                    || BigDecimal.ONE.compareTo(sumBlockFactors) == 0;

            blockWrapper.getFactor()
                .map(factor -> sumIsZeroOrOne ? factor : factor.divide(sumBlockFactors))
                .ifPresent(q -> this.quantity = q.doubleValue());
```

File: glsk/glsk-document-io-api/src/main/java/com/powsybl/glsk/api/io/GlskDocumentImporter.java (hunk 1 above)
Comment (schwitzguebel) [reply glsk/glsk-document-io-api/src/main/java/com/powsybl/glsk/api/io/GlskDocumentImporter.java:29]: Sorry I forgot that one, I will even add two methods ;) one for validate and on for validate with calculation direction

**Review by vbochetRTE — APPROVED** [review]

**Review by phiedw — APPROVED** [review]

**Pull request merged**